ip_test.go

```diff
@@ -24,7 +24,7 @@ func TestIP(t *testing.T) {
 		{"1.2.3.4", true, "1.2.3.4"},
 		{"127.0.0.1", true, "127.0.0.1"},
 		{"255.255.255.255", true, "255.255.255.255"},
-		{"", false, ""},
+		{"", true, "0.0.0.0"},
```

Comment:
Why is the nil value of an ip address 0.0.0.0?  How would a user distinguish between explicitly set 0.0.0.0, and no ip address specified?

Reply:
Oh, the way this test is constructed the default flag value is defined [here](https://github.com/spf13/pflag/blob/66d8d2fb58639fbd8ee3b982820089830976a2a9/ip_test.go#L12),  what this test is saying is that if we're given a blank string, return the default value which in this case happens to be `0.0.0.0`. This is not to say always return `0.0.0.0` for all blank string inputs.

Reply:
gotcha, thx for clarifying.